src/coreclr/jit/forwardsub.cpp

```diff
@@ -0,0 +1,622 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "jitpch.h"
+
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+// Simple Forward Substitution
+
+//------------------------------------------------------------------------
+// fgForwardSub: try forward substitution in this method
+//
+// Returns:
+//   suitable phase status
+//
+PhaseStatus Compiler::fgForwardSub()
+{
+    bool changed = false;
+    bool enabled = true;
+
+#if defined(DEBUG)
+    enabled = JitConfig.JitNoForwardSub() == 0;
```

Comment:
Wouldn't this be simpler to allow early-exit, and get rid of `enabled` on the extra nesting it induces?
```suggestion
    if (JitConfig.JitNoForwardSub())
    {
        return PhaseStatus::MODIFIED_NOTHING;
    }
```